tt_metal/impl/debug/watcher_server.cpp

```diff
@@ -539,16 +539,22 @@ static void dump_core(
     validate_kernel_ids(f, used_kernel_names, core, &mbox_data->launch);
 
     if (watcher::enabled) {
+        auto &disabled_features_set = tt::llrt::OptionsG.get_watcher_disabled_features();
         // Dump state only gathered if device is compiled w/ watcher
-        dump_debug_status(f, core, &mbox_data->launch, mbox_data->debug_status);
+        if (disabled_features_set.find("STATUS") == disabled_features_set.end())
+            dump_debug_status(f, core, &mbox_data->launch, mbox_data->debug_status);
         // Ethernet cores have firmware that starts at address 0, so no need to check it for a
         // magic value.
         if (!is_eth_core)
             dump_l1_status(f, device, core,  &mbox_data->launch);
-        dump_noc_sanity_status(f, core, &mbox_data->launch, mbox_data->sanitize_noc, mbox_data->debug_status);
-        dump_assert_status(f, core, &mbox_data->launch, &mbox_data->assert_status, mbox_data->debug_status);
-        dump_pause_status(core, &mbox_data->pause_status, paused_cores);
-        dump_ring_buffer(f, device, core);
+        if (disabled_features_set.find("NOC_SANITIZE") == disabled_features_set.end())
```

Comment:
did you see a way to avoid having magic strings in more than one place? I can pull the strings out of the watcher server code, but I'm not seeing a way to avoid needing them in both rtoptions and the #defines